core/config/yaml_test.go

```diff
@@ -146,3 +146,34 @@ func TestMatchPopulatedEmptyStruct(t *testing.T) {
 	empty := reflect.New(reflect.TypeOf(es)).Elem().Interface()
 	assert.True(t, reflect.DeepEqual(empty, es))
 }
+
+type pointerStruct struct {
+	MyTrueBool  *bool   `yaml:"myTrueBool"`
+	MyFalseBool *bool   `yaml:"myFalseBool"`
```

Comment:
Yes, there is a separate test for checking nil *bools and other types on empty yaml.